### ISSUE TITLE
Swift dynamic cast failure

### ISSUE THREAD

**Issue by dirkfabisch** [QueryKit/QueryKit#22]
When I try to us the non NSPredicate notation I get an `Swift dynamic cast failure`.

```
#1	0x0000000103153d7e in QueryKit.< infix <A>(QueryKit.Attribute<A>, A) -> ObjectiveC.NSPredicate at /Users/dirk/Development/Test_App/dummyDB/Pods/QueryKit/QueryKit/Attribute.swift:59
```

Working code
    
    var predicate = NSPredicate(format: "age == 42")
    var qs2 = Person.queryset(moc).filter(predicate!)

Error code

    var qs = Person.queryset(moc).filter(Person.attributes.age == 42)

Model is 

```
Entity:Person
Attribute:age::NSNumber
Attribute:name::String
```

Build with mogenerator template.

**Comment by kylef** [comment]
@dirkfabisch Just to check, is your attribute for age an optional?

```swift
var age:Attribute<NSNumber?> {
  return Attribute< NSNumber?>("age")
}
```

vs

```swift
var age:Attribute<NSNumber> {
  return Attribute< NSNumber>("age")
}
```

**Comment by dirkfabisch** [comment]
It is an optional attribute

       var age:Attribute<NSNumber?> {
            return Attribute<NSNumber?>("age")
        }
 
The complete repository you will find here: (https://github.com/dirkfabisch/dummyDB)

**Comment by kylef** [comment]
@dirkfabisch Thanks for reporting this, me and @neonichu got to the bottom of this problem and I've released QueryKit 0.9.1 with a fix of using optional types with Attribute.

**Comment by wodetian** [comment]
I think you did a very good thing, thank you very much ~ 🍻🍻🍻
